### PR TITLE
Implement DateCreated column

### LINKED ISSUE #3189: Sort content by creation date.
****Is your feature request related to a problem? Please describe.**
User Experience

**Describe the solution you'd like**
Sort the columns by Creation Date. Add to the list the option to sort by Creation Date and that the option can be added from the right button on the column titles.

**Describe alternatives you've considered**
Alternatively, add the option from Sort By to the right button over any area where it is clicked.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by BrianAllred**
Fixes #3189
All the pieces for this are here as far as I can tell, but I'm still missing something. The sorting doesn't work correctly - items change order, but not in any logical fashion. I'm submitting this as a draft so I can get some feedback, any would be appreciated.

### CODE REVIEW

**Comment by gave92** [comment]
@BrianAllred you simply missed to add date created to: `DataGridColumn SortedColumn`

File: Files/Strings/en-US/Resources.resw [hunk 1]
```diff
@@ -1851,4 +1851,10 @@
   <data name="DriveTypeVirtualDrive" xml:space="preserve">
     <value>Virtual Drive</value>
   </data>
+  <data name="BaseLayoutContextFlyoutSortByDateCreated.Text" xml:space="preserve">
+    <value>Date created</value>
+  </data>
+  <data name="dateCreatedColumn.Header" xml:space="preserve">
```
Comment (yaichenbaum) [inline Files/Strings/en-US/Resources.resw:1857]:
```suggestion
  <data name="DateCreatedColumn.Header" xml:space="preserve">
```

File: Files/Views/LayoutModes/GenericFileBrowser.xaml [hunk 2]
```diff
@@ -958,12 +976,32 @@
                         <ToolTip Content="{Binding FolderTooltipText}" />
                     </ToolTipService.ToolTip>
                 </controls:DataGridTextColumn>
+                <controls:DataGridTextColumn
+                    x:Name="dateCreatedColumn"
+                    x:Uid="dateCreatedColumn"
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GenericFileBrowser.xaml:981]:
```suggestion
                    x:Uid="DateCreatedColumn"
```

File: Files/ViewModels/SettingsViewModel.cs [hunk 3]
```diff
@@ -269,6 +269,15 @@ public bool ShowDateColumn
             set => Set(value);
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether or not the date created column should be visible.
+        /// </summary>
+        public bool ShowDateCreatedColumn
+        {
+            get => Get(true);
```
Comment (yaichenbaum) [inline Files/ViewModels/SettingsViewModel.cs:277]:
```suggestion
            get => Get(false);
```

File: Files/ViewModels/SettingsViewModel.cs [hunk 4]
```diff
@@ -269,6 +269,15 @@ public bool ShowDateColumn
             set => Set(value);
         }
```
Comment (yaichenbaum) [inline Files/ViewModels/SettingsViewModel.cs:270]: Should this setting be renamed to better indicate what it does?

**Review by yaichenbaum — APPROVED** [review]
Great work! LGTM

**Pull request merged**